sql/moz-fx-data-shared-prod/firefox_desktop_derived/newtab_visits_daily_v2/query.sql

```diff
@@ -0,0 +1,284 @@
+  LOGICAL_OR(event_category = 'topsites' AND event_name IN ('click', 'dismiss'))
+  AND LOGICAL_OR(is_default_ui) AS is_topsites_interaction,
+  LOGICAL_OR(event_category = 'topsites' AND event_name IN ('click'))
+  AND LOGICAL_OR(is_default_ui) AS is_topsites_click,
+  LOGICAL_OR(event_category = 'topsites' AND event_name IN ('impression'))
+  AND LOGICAL_OR(is_default_ui) AS is_topsites_impression,
+  LOGICAL_OR(
+    event_category = 'topsites'
+    AND event_name IN ('click', 'dismiss')
+    AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_sponsored_topsites_interaction,
+  LOGICAL_OR(
+    event_category = 'topsites'
+    AND event_name IN ('click')
+    AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_sponsored_topsites_click,
+  LOGICAL_OR(
+    event_category = 'topsites'
+    AND event_name IN ('impression')
+    AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_sponsored_topsites_impression,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(event_category = 'topsites' AND event_name IN ('click')),
+    0
+  ) AS any_topsites_click_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(event_category = 'topsites' AND event_name IN ('impression')),
+    0
+  ) AS any_topsites_impression_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('click')
+      AND NOT SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS organic_topsites_click_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('impression')
+      AND NOT SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS organic_topsites_impression_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('click')
+      AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS sponsored_topsites_click_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('impression')
+      AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS sponsored_topsites_impression_count,
+  LOGICAL_OR(
+    event_category = 'newtab'
+    AND event_name IN (
+      'weather_change_display',
+      'weather_open_provider_url',
+      'weather_location_selected'
+    )
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_widget_interaction,
+  LOGICAL_OR(
+    event_category = 'newtab'
+    AND event_name IN (
+      'wallpaper_click',
+      'wallpaper_category_click',
+      'wallpaper_highlight_cta_click',
+      'wallpaper_highlight_dismissed'
+    )
+  )
```

Comment:
Consolidate `LOGICAL_OR`, see comment for line 95

Reply:
I have explained it [here](https://github.com/mozilla/bigquery-etl/pull/7362/files#r2093337681)